### ISSUE TITLE
What is the purpose of kn route list <name> ?

### ISSUE THREAD

**Issue by mgencur** [knative/client#292]
## Ask your question here:
What is the purpose of `kn route list <route_name>` ? Should we drop this option?

The point is that `kn route list <route_name>` doesn't give us anything on top of the usual `kn route list`. It would make sense if we were listing routes for a given service but here we're passing the name of the route itself. When we want to get more details about the route we can use `kn route describe <route_name>` 
I suppose similar concerns apply for `kn service list <service_name>` as well

**Comment by mgencur** [comment]
One more point: It's confusing to me to call a "list" operation on a single object and as a result have a list with a single object

**Comment by rhuss** [comment]
At the moment only `kn <noun> list -o yaml` is supposed to provide machine readable output. That `kn <nound> describe` currently spits out yaml is a placeholder. It will provide human readable output (like `kubectl describe`). It could be that `describe` will get a `-o` option, too (`kubectl describe` doesn't have that), but that still open.

Besides that, I doubt the usefulness of the `kn route` commands in general (route listing happens alos via `kn service describe` and you won't be able to create/update routes via `kn` anyway).

**Comment by navidshaikh** [comment]
Also `kn <resource> list <resource_name> -o yaml` returns the YAML of `kind: <resource>List` but may be we want the return YAML of `kind: <resource>`.

**Comment by rhuss** [comment]
> Also `kn <resource> list <resource_name> -o yaml` returns the YAML of `kind: <resource>List` but may be we want the return YAML of `kind: <resource>`.

Right, that's the main motivation to allow a `-o yaml` for `kn service describe`, too. IMO that's better than the kubectl way where you get either a list or single object depending how you call `kubectl get`.

**Comment by rhuss** [comment]
> One more point: It's confusing to me to call a "list" operation on a single object and as a result have a list with a single object

yes, agreed. I would love to have this seen as a 'filter' with a 'startsWith'  semantics, not only for an exact match. This would help especially for `kn revision list myservice` as revisions usually have the service as prefix and a random suffix.

**Comment by navidshaikh** [comment]
Route list with an argument is following other resources' list commands. My cluster may have a 10+ services/routes and I list the different configs of my service by listing commands, I would do
```
16:00 ➜  /tmp  kn service list hello
NAME    URL                                     GENERATION   AGE     CONDITIONS   READY   REASON
hello   http://hello.default.apps-crc.testing   3            2d19h   3 OK / 3     True    

16:00 ➜  /tmp  kn revision list -s hello
NAME            SERVICE   GENERATION   AGE     CONDITIONS   READY   REASON
hello-v1        hello     2            94m     3 OK / 4     True    
hello-jvjpp-1   hello     1            2d19h   3 OK / 4     True    

16:00 ➜  /tmp  kn route list hello
NAME    URL                                     AGE     CONDITIONS   TRAFFIC
hello   http://hello.default.apps-crc.testing   2d19h   3 OK / 3     100% -> hello-jvjpp-1
```
if route list were to print the routes for all the services, it'd be hard to parse.

**Comment by navidshaikh** [comment]
Closing the issue, feel free to open if there is a question.
